macros/timeadd.sql

```diff
@@ -0,0 +1,31 @@
+{%- macro timeadd(part, amount_to_add, value) -%}
+    {# adds `amount_to_add` `part`s to `value`. so adding one hour to Jan 1 2020 01:00:00 would be timeadd('hour',1,'2020-01-01 01:00:00'). 
+       NOTE: dateadd only manipulates time values. for date additions see [dateadd](#dateadd) 
+       ARGS:
+         - part (string) one of 'second','minute','hour'.  
+         - amount_to_add (int) number of `part` units to add to `value`. Negative subtracts.
+         - value (string) the date time string or column to add to.
+       RETURNS: a date time value with the amount added.
+    #}
+    {%- set part = part |lower -%}
+    {%if part not in ('second','minute','hour') %}
+        {% if part in ('day','week','month','year') %}
+            {{exceptions.raise_compiler_error("date component passed to macro `timeadd()`. Did you want `dateadd()`?")}}
```

Comment:
this is beautiful

Reply:
Do I point out that I stole this directly from dateadd which @norton120  wrote, or just take the compliment?

Reply:
I knew I liked that style..